### PR TITLE
Modified PascalCaseWordPartsRegex to accept more "word characters"

### PR DESCRIPTION
**Pull request by hangy**
Instead of [A-Z] this uses \p{Lu} (Upper Case Unicode letters) and instead of [a-z] this uess \p{Ll} (lower case Unicode letters). This enables us to use some word characters of different languages, see #532 and 563.

Note that this does not handle the example "mühldorfer STRAßE 23" from #532, but if it were written as "mühldorfer STRAẞE 23", it would be okay.  The previously missing capital ß (ẞ) would require special culture dependent handling, which I do not know if it is wanted in here. However, this deals with "MÜNCHEN" and "ajouté" fine and better than the previous version.

The current `StringHumanizeTests` pass this change, but there might be unexpected changes in production code, as much, much more characters will be accepted and returned in the humanized string.

---
This change is [Reviewable](https://reviewable.io/reviews/humanizr/humanizer/564)

### CODE REVIEW

**Comment by jtheisen** [comment]
Great! Hope this gets accepted.

**Comment by hazzik** [comment]
The changes LGTM, but can you please add some tests to cover the existing issues, #532 for ex.?

**Comment by onovotny** [comment]
@hazzik is this ready to merge?

**Comment by hazzik** [comment]
Yep

File: src/Humanizer.Tests.Shared/StringHumanizeTests.cs [hunk 1]
```diff
@@ -18,11 +18,20 @@ public class StringHumanizeTests
...
+        [InlineData("JeNeParlePasFrançais", "Je ne parle pas français")]
         public void CanHumanizeStringInPascalCase(string input, string expectedResult)
         {
             Assert.Equal(expectedResult, input.Humanize());
         }
 
+        [Theory, UseCulture("tr-TR")]
+        [InlineData("istanbul", "İstanbul")]
+        [InlineData("diyarbakır", "Diyarbakır")]
+        public void CanHumanizeStringInPascalCaseInTurkish(string input, string expectedResult)
```
Comment (hazzik) [inline src/Humanizer.Tests.Shared/StringHumanizeTests.cs]: :+1:

**Pull request merged**